### PR TITLE
feat: improve badge component

### PR DESCRIPTION
**Pull request by bjohnsonrmn**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.

---------
**Outline your feature or bug-fix below**

My code improves the Badge component by:
* Adding props to make the size, colors, and positioning configurable
* Adding stories to give examples of options
* Creating snapshot and unit tests
* Filling out the README with info on usage and props

### CODE REVIEW

**Review by sharkparty — CHANGES_REQUESTED** [review]
Can you make sure the CI passes?

**Comment by bjohnsonrmn** [comment]
@sharkparty yes, let me see if I can get the CI to pass. It seems that it is failing on global branch test coverage.

**Comment by bjohnsonrmn** [comment]
@sharkparty CI is passing 🎉 

I had to clean up a few linting errors in `Badge` and I also somehow picked up a `Button` change that I needed to remove.

File: src/Badge/Badge.component.tsx [hunk 1]
```diff
@@ -2,45 +2,148 @@
...
+    display: flex;
     font-weight: 700;
-    font-size: 0.75rem;
-    padding: 1px ${sizes.padding.xs};
-    border-radius: ${variables.borderRadius};
+    justify-content: center;
+    text-align: center;
+
+    /* Border Radius */
+    ${({ count = 0, size = 'dot' }: BadgeProps) =>
```
Comment (bjohnsonrmn) [inline src/Badge/Badge.component.tsx]: The linter is saying these lines are uncovered:
```
 70,76,86 
```

I'm not sure how to test those. Any ideas?

**Review by sharkparty — APPROVED** [review]
LGTM

File: src/Badge/Badge.component.tsx (hunk 1 above)
Comment (sharkparty) [reply src/Badge/Badge.component.tsx]: Coverage reports all have their anomalies. Likely it'd be more difficult to test this than the value that would be offered by the test. I typically focus on gaining coverage of other uncovered JS files instead of trying to endlessly chase a few lines that I'm pretty confident in. In order to test these you'd need to mock both forks of the logic in your ternary inside of a spec.

File: src/Badge/Badge.component.tsx [hunk 2]
```diff
@@ -2,45 +2,148 @@
...
-        <StyledBadge>{count || '•'}</StyledBadge>
-    </StyledBadgeContainer>
-);
+}: BadgeProps): JSX.Element => {
+    const getDisplayCount = (
+        badgeCount: number,
+        badgeOverflowCount: number
+    ) => {
+        if (badgeCount === 0) {
+            return '';
```
Comment (sharkparty) [inline src/Badge/Badge.component.tsx:121]: This was a desired behavior: '•'

**Review by sharkparty — CHANGES_REQUESTED** [review]

File: src/Badge/Badge.component.tsx (hunk 1 above)
Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx]: @sharkparty that makes sense. I think I have tests for the forks of the ternary, just not the assignment of the defaults. Could you remind me what the longhand of this is? 
```
 ${({ count = 0, size = 'dot' }: BadgeProps) =>
```

Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx]: @sharkparty that makes sense. I think I have tests for the forks of the ternary, just not the assignment of the defaults. Could you remind me what the longhand of this is?

``` 
${({ count = 0, size = 'dot' }: BadgeProps) =>
```

File: src/Badge/Badge.component.tsx (hunk 2 above)
Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx:121]: Let me sync with Jen on this. I think she said she didn't want that, though let's make sure we have all the cases covered.

Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx:121]: @sharkparty I talked with Jen and she said she does not want a `•`.  

Our conversation did bring up an uncovered case though. She wants the default to be that the Badge does not show if the `count` is `0`, but she also wants the option to render a Badge with a value of `0`, so I'm planning to add a `showZero` prop that defaults to `false` like Ant does.

Comment (sharkparty) [reply src/Badge/Badge.component.tsx:121]: Engineering wants the ability to use that • as a value. So long as that is possible, I'm cool with this implementation.

**Comment by sharkparty** [comment]
@bjohnsonrmn I just rebased this against master. Make sure to pull before changing anything.

File: src/Badge/Badge.component.tsx (hunk 2 above)
Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx:121]: Sure, we can make that possible. Can you tell me more about when we would use `•` instead of a number? I'm wondering if I need to connect it to a certain state of the component or just display it correctly if it is passed in via the `count` prop.

Comment (sharkparty) [reply src/Badge/Badge.component.tsx:121]: It was a request made by Andy during the UI Guild that I added. I think it's how we show that badge on RMN core if we don't have a count available but do know that there is a notification. I really didn't push back on it bc it was pretty trivial to add.

Comment (bjohnsonrmn) [reply src/Badge/Badge.component.tsx:121]: Hey Kam, I just chatted with Andy and he said that using the dot size of the badge would work for the use case you mentioned. Are you ok with that solution?

Comment (sharkparty) [reply src/Badge/Badge.component.tsx:121]: Absolutely. He’s the stakeholder, so if he opts into something else then that’s great!

**Review by sharkparty — APPROVED** [review]
LGTM x 2

**Review by sharkparty — APPROVED** [review]
👍

**Review by sharkparty — APPROVED** [review]
